tests/datapoint_serialization.cpp

```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+
+#include "Datapoint.h"
+
```

Comment:
I think it's good to have a unit test like this to capture intended serialization / deserialization behavior.

I think the assertions I wrote are clumsy, and we might prefer to adopt [Google Test](https://google.github.io/googletest/quickstart-cmake.html) or similar.